### PR TITLE
feat(argocd-apps): Add rollout strategy for ApplicationSet

### PR DESCRIPTION
**Pull request by khuedoan**
[Progressive syncs](https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Progressive-Syncs/) is a new experimental feature. [It must be explicitly enabled](https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Progressive-Syncs/#enabling-progressive-syncs), so I'm not sure if we should include it in the default `values.yaml` yet.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Comment by yu-croco** [comment]
Hi @khuedoan , thank you for your PR.
Can you please also add the value on [values.yaml](https://github.com/argoproj/argo-helm/blob/main/charts/argocd-apps/values.yaml)?

**Comment by khuedoan** [comment]
Hi @yu-croco, thank you for reviewing! Since this feature is still behind a feature gate and I'm not sure whether the specification is gonna change, is a basic example like the one below sufficient? Or should I add a [full example](https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Progressive-Syncs/#example))?

```yaml
strategy:
    type: AllAtOnce
```

**Comment by yu-croco** [comment]
Hi @khuedoan , thank you for your reply.
I'm sorry I didn't check it is still alpha. 😇 

We usually provide only beta features, so I wonder we can add it... 🤔 WDYT?
@mkilchhofer @jmeridth @tic

**Comment by mkilchhofer** [comment]
> We usually provide only beta features, so I wonder we can go proceed this PR... 🤔 WDYT? @mkilchhofer @jmeridth @tic *We usually put `awaiting-upstream` label to the PR and wait until the feature is GA.

I am okay with adding it as the argocd-apps chart is some kind of a "metachart" only 👍 .

**Comment by yu-croco** [comment]
Thank you for your opinion @mkilchhofer .

>as the argocd-apps chart is some kind of a "metachart" only 👍 .

You're right. I agree with this.  🚀 


@khuedoan 
> I'm not sure whether the specification is gonna change, is a basic example like the one below sufficient? Or should I add a [full example](https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Progressive-Syncs/#example)?

I think we can add the full example at `applicationsets: []` block with the doc URL. 🙋

**Comment by khuedoan** [comment]
Thank you! I've added the full example and a test value file as well.

File: charts/argocd-apps/ci/applicationsets-progressive-syncs-values.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Test applicationsets with Progressive Syncs
+
+applicationsets:
```
Comment (yu-croco) [inline charts/argocd-apps/ci/applicationsets-progressive-syncs-values.yaml:3]: 👍

**Review by yu-croco — APPROVED** [review]
Thank you for your contribution! LGTM!

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**